atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/TargetedSweeper.java

```diff
@@ -241,6 +254,45 @@ private int getPartitionBatchLimit() {
                 : runtime.get().maximumPartitionsToBatchInSingleRead();
     }
 
+    @VisibleForTesting
+    Void updateLastSweptTs(SweeperStrategy sweeperStrategy) {
+        int shards = queue.getNumShards();
+
+        Set<ShardAndStrategy> shardAndStrategySet = IntStream.range(0, shards)
+                .mapToObj(shard -> ShardAndStrategy.of(shard, sweeperStrategy))
+                .collect(Collectors.toSet());
+
+        Map<ShardAndStrategy, Long> shardAndStrategyToTimestamp = queue.getLastSweptTimestamps(shardAndStrategySet);
+
+        KeyedStream.stream(shardAndStrategyToTimestamp)
+                .filter(lastSweptTimestamp -> lastSweptTimestamp != -1L)
```

Comment:
This will be a change w.r.t. previous behaviour, but we don't actually want to filter these out -- if a shard is stuck at -1, we should report it.